android-agent/src/main/java/io/opentelemetry/android/internal/services/ServiceManager.java

```diff
@@ -25,9 +26,14 @@ public final class ServiceManager implements Lifecycle {
 
     @RestrictTo(RestrictTo.Scope.LIBRARY_GROUP_PREFIX)
     public static void initialize(Context appContext) {
+        if (instance != null) {
+            // Already initialized.
+            return;
+        }
         instance = new ServiceManager();
         instance.addService(PreferencesService.create(appContext));
         instance.addService(new CacheStorageService(appContext));
+        instance.addService(new PeriodicWorkService());
```

Comment:
This is technically a functional change, do we want a test around this?

Reply:
I've added some validations for the registered services. It was easier by making it a robolectric test though so I had to change it to junit4 to make it work.